.github/workflows/run-pytest.yml

```
@@ -0,0 +1,66 @@
+name: Run Unit Tests
+
+on:
+  # trigger on pull requests
+  pull_request:
+
+  # trigger on all commits to master
+  push:
+    branches:
+      - 'master'
```

```suggestion
      - 'master'
      - 'main'
```
If/when we migrate from master to main as our default branch, this will help future-proof it (but we will have to update all the badges so as long as you remember to update the workflow as well, this is less useful)

---

We'll be doing a mass find-replace when we make that branch transition, so I'm going to hold off until then.